app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarDetailsFragment.kt

```diff
@@ -61,48 +69,57 @@ class AddCalendarDetailsFragment: Fragment() {
-                    requireActivity().finish()
-                true
-            } else
-                false
+        if (item.itemId == R.id.create_calendar) {
+            CoroutineScope(Dispatchers.IO).launch { createCalendar() }
+            true
+        } else
+            false
 
 
-    private fun createCalendar(): Boolean {
+    @WorkerThread
+    private suspend fun createCalendar() {
         val account = AppAccount.get(requireActivity())
 
-        val calInfo = ContentValues(9)
-        calInfo.put(Calendars.ACCOUNT_NAME, account.name)
-        calInfo.put(Calendars.ACCOUNT_TYPE, account.type)
-        calInfo.put(Calendars.NAME, titleColorModel.url.value)
-        calInfo.put(Calendars.CALENDAR_DISPLAY_NAME, titleColorModel.title.value)
-        calInfo.put(Calendars.CALENDAR_COLOR, titleColorModel.color.value)
-        calInfo.put(Calendars.OWNER_ACCOUNT, account.name)
-        calInfo.put(Calendars.SYNC_EVENTS, 1)
-        calInfo.put(Calendars.VISIBLE, 1)
-        calInfo.put(Calendars.CALENDAR_ACCESS_LEVEL, Calendars.CAL_ACCESS_READ)
-        calInfo.put(LocalCalendar.COLUMN_IGNORE_EMBEDDED, titleColorModel.ignoreAlerts.value)
-        calInfo.put(LocalCalendar.COLUMN_DEFAULT_ALARM, titleColorModel.defaultAlarmMinutes.value)
-
-        val client: ContentProviderClient? = requireActivity().contentResolver.acquireContentProviderClient(CalendarContract.AUTHORITY)
-        return try {
-            client?.let {
-                val uri = AndroidCalendar.create(account, it, calInfo)
-                val calendar = LocalCalendar.findById(account, client, ContentUris.parseId(uri))
-
-                if (credentialsModel.requiresAuth.value == true)
-                    CalendarCredentials(requireActivity()).put(calendar, credentialsModel.username.value, credentialsModel.password.value)
+        try {
+            val subscription = Subscription(
+                id = 0L,
+                eTag = null,
+                lastModified = 0L,
+                lastSync = 0L,
+                url = Uri.parse(titleColorModel.url.value!!),
+                displayName = titleColorModel.title.value!!,
+                color = titleColorModel.color.value,
+                ignoreEmbeddedAlerts = titleColorModel.ignoreAlerts.value ?: false,
+                defaultAlarmMinutes = titleColorModel.defaultAlarmMinutes.value,
+                accountName = account.name,
+                accountType = account.type,
+            )
+
+            if (credentialsModel.requiresAuth.value == true)
+                CalendarCredentials(requireActivity()).put(subscription, credentialsModel.username.value, credentialsModel.password.value)
+
+            Log.v(TAG, "Adding subscription to database...")
+            AppDatabase.getInstance(requireContext())
+                .subscriptionsDao()
+                .add(subscription)
+
+            Log.v(TAG, "Adding subscription to system...")
+            subscription.createAndroidCalendar(requireContext())
```

Comment:
`subscription.create` should be used instead of manual creation and addition to the database.